x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -159,14 +151,39 @@ class AgentPolicyService {
     soClient: SavedObjectsClientContract,
     esClient: ElasticsearchClient,
     newAgentPolicy: NewAgentPolicy,
-    searchParams: {
-      searchFields: string[];
-      search: string;
-    }
+    searchParams:
+      | { id: string }
+      | {
+          searchFields: string[];
+          search: string;
+        }
   ): Promise<{
     created: boolean;
     policy: AgentPolicy;
   }> {
+    // For preconfigured policies with a specified ID
+    if ('id' in searchParams) {
+      try {
+        const agentPolicy = await soClient.get<AgentPolicySOAttributes>(
+          AGENT_POLICY_SAVED_OBJECT_TYPE,
+          searchParams.id
+        );
+        return {
+          created: false,
+          policy: {
+            id: agentPolicy.id,
+            ...agentPolicy.attributes,
+          },
+        };
+      } catch (e) {
+        return {
```

Comment:
Yeah that's the deal, if it can't find the requested ID it 404s.

Reply:
But if it throws for another reason, we'll still proceed. How about we guard this to only try if the SO isn't found, and otherwise throw or do something to indicate we're in an unexpected condition? We can use an SO helper like

https://github.com/elastic/kibana/blob/cb3c4e3a212255a9e9b8c89e784e0e452b661233/x-pack/plugins/fleet/server/routes/package_policy/handlers.ts#L67